### PR TITLE
Code signing

### PR DESCRIPTION
**Pull request by jhreis**
- Code signing based on xcconfig files
- Setup for easy local development (e.g. local signing without git conflicts)
- Bootstrap script setup to copy over any new xcconfig files needed for local development
- Archive process works with Brave code signing

### CODE REVIEW

File: Client/Configuration/Fennec.xcconfig [hunk 1]
```diff
@@ -2,19 +2,28 @@
...
+
 // Adjust SDK Variables
 ADJUST_ENVIRONMENT =
 ADJUST_APP_TOKEN =
 
 // Leanplum SDK Variables
 LEANPLUM_ENVIRONMENT = development
 LEANPLUM_APP_ID = 
-LEANPLUM_KEY =
+LEANPLUM_KEY =
```
Comment (kylehickinson) [inline Client/Configuration/Fennec.xcconfig]: Leanplum SDK variables can be removed (doesn't have to be done in this PR)

**Review by kylehickinson — COMMENTED** [review]
Was slightly confused about how to get it all setup, but that can be mitigated by updating `README` to specify that _after_ you run `bootstrap.sh` you have to update `Client/Configurations/Local/DevTeam.xcconfig` to finish.

**Comment by jhreis** [comment]
To test, use a fresh or previous clone:

- run bootstrap.sh as normal (will need to re-run after checking out this branch)
- go into Client/Configurations/Local/DevTeam.xcconfig
- add personal dev team

Project should now code sign `fennec` and compile fine. The last two steps are only necessary for running on device (as sims don't need to be signed).

Other schemes will not sign with personal dev team, as those are set to use the Brave team.

**Review by kylehickinson — APPROVED** [review]
Works well once you figure out your Team ID.

Documentation around usage and possibly some for obtaining your Team ID would be great but we can add that before open sourcing

File: fastlane/Fastfile [hunk 2]
```diff
@@ -1,26 +1,126 @@
...
+      sdk: "iphoneos",
+      clean: true,
+      export_method: "app-store",
+      include_symbols: true,
+      include_bitcode: false,
+      xcargs: "-allowProvisioningUpdates",
+    )
+
+    testflight(
+      username: sh("cat", "../Client/Configuration/Local/AppleId"),
```
Comment (iccub) [inline fastlane/Fastfile]: I would extract `../Client/Configuration/` to a `CONFIG_PATH ` constant, it's being used in few places.
We do that in `bootstrap.sh` already.

File: fastlane/Fastfile [hunk 3]
```diff
@@ -1,26 +1,126 @@
...
+      ensure_git_status_clean()
+    end
+  end
 
-import_build_tools(
-  url: 'git@github.com:mozilla/firefox-ios-build-tools.git',
-  clone_folder: "../build-tools",
-)
+  after_all do |lane, options|
+    # If no uncommitted changes existed before running `lane`, reset any changes potential fastlane changes
```
Comment (iccub) [inline fastlane/Fastfile]: `reset any changes potential fastlane changes` to `reset any potential fastlane changes`

File: fastlane/Fastfile [hunk 4]
```diff
@@ -1,26 +1,126 @@
...
+    unless options[:ignore_git_status]
```
Comment (iccub) [inline fastlane/Fastfile]: Can be inlined to `reset_git_repo(skip_clean: true) unless options[:ignore_git_status]`. Not a biggie, it's just a script

File: Client/Configuration/Base.xcconfig [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+#include "Local/BuildId.xcconfig"
+
+OTHER_SWIFT_FLAGS_BASE=-DMOZ_TARGET_$(TARGET_NAME:upper)
+
+FRAMEWORK_BASE_BUNDLE_ID = com.brave
+BASE_BUNDLE_ID = $(FRAMEWORK_BASE_BUNDLE_ID).ios
+
+DEVELOPMENT_TEAM = KL8N8XSYF4
+
+BRAVE_VERSION = 2.0
```
Comment (iccub) [inline Client/Configuration/Base.xcconfig:19]: Is it a manual job to set new `BRAVE_VERSION` for each new build or does fastlane do it for us?

File: Client/Configuration/Firefox.xcconfig [hunk 6]
```diff
@@ -2,19 +2,16 @@
...
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-MOZ_BUNDLE_DISPLAY_NAME = Firefox
+#include "Base.xcconfig"
+
+MOZ_BUNDLE_DISPLAY_NAME = Brave
 
 // Bundle Identifier
-MOZ_BUNDLE_ID = org.mozilla.ios.Firefox
+MOZ_BUNDLE_ID = $(BASE_BUNDLE_ID).Browser3
```
Comment (iccub) [inline Client/Configuration/Firefox.xcconfig]: Is `Browser3` our bundle id or just a temporary name?

File: Client/Entitlements/FirefoxApplication.entitlements [hunk 7]
```diff
@@ -2,24 +2,15 @@
...
-		<string>applinks:nightly.dev.lcip.org</string>
-		<string>applinks:rzte.adj.st</string>
-		<string>applinks:nn8g.adj.st</string>
-	</array>
 	<key>aps-environment</key>
 	<string>production</string>
 	<key>com.apple.security.application-groups</key>
 	<array>
-		<string>group.org.mozilla.ios.Firefox</string>
+		<string>group.come.brave.ios.Firefox</string>
```
Comment (iccub) [inline Client/Entitlements/FirefoxApplication.entitlements]: Maybe remove that `.Firefox` for now, same in few other places.

File: fastlane/Fastfile (hunk 4 above)
Comment (jhreis) [reply fastlane/Fastfile]: Oh this is nice. I totally forgot this was a possibility! Not too familiar with Ruby 😄

File: Client/Configuration/Base.xcconfig (hunk 5 above)
Comment (jhreis) [reply Client/Configuration/Base.xcconfig:19]: Yes, currently manual. May adjust this, but going to leave it manual for now. Will probably move to a lane in the future though.

File: Client/Entitlements/FirefoxApplication.entitlements (hunk 7 above)
Comment (jhreis) [reply Client/Entitlements/FirefoxApplication.entitlements]: I renamed some of the groups, but didn't both with adjusting the Firefox/Fennec name spaces here. These are for app groupings and will re-address these when needed. I did fix the type on the prefix though 😄

**Review by iccub — APPROVED** [review]

**Pull request merged**